### PR TITLE
feat: implement key cache

### PR DESCRIPTION
**Pull request by neekolas**
## What does this PR do?

- Caches keys in `localStorage` to avoid repeated signature requests
- I've done an audit of the third party libraries used in Lenster to see if anything is uploading LocalStorage contents to a third party. Looks all clear. But we would have to be mindful of this moving forwards.
- Wipes keys from LocalStorage on logout

Fixes # (issue)

https://user-images.githubusercontent.com/65710/196823874-9c2f1d49-cd13-4c65-b89b-23fea88c772e.mp4

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Sign in to the messaging page by approving the signature request in your wallet
- [ ] Refresh the page. No signature request should be present
- [ ] Navigate between pages, still no signature request should be present
- [ ] Logout and log back in to Lenster. Should be presented with a signature request the next time you reach the Messaging page.

### CODE REVIEW

File: src/components/utils/hooks/useXmtpClient.tsx [hunk 1]
```diff
@@ -5,6 +5,19 @@ import { useAppStore } from 'src/store/app';
...
+
+const buildLocalStorageKey = (walletAddress: string) => `xmtp:keys:${walletAddress}`;
+
+const loadKeys = (walletAddress: string): Uint8Array | null => {
+  const val = localStorage.getItem(buildLocalStorageKey(walletAddress));
+  return val ? Buffer.from(val, ENCODING) : null;
+};
+
+const storeKeys = (walletAddress: string, keys: Uint8Array) => {
+  localStorage.setItem(buildLocalStorageKey(walletAddress), Buffer.from(keys).toString(ENCODING));
```
Comment (elisealix22) [inline src/components/utils/hooks/useXmtpClient.tsx:18]: Can we clear the localStorage on logout? I think currently we're listening for changes in `currentProfile` to determine whether to reset the state or not in other hooks.

**Review by elisealix22 — COMMENTED** [review]
Game changer. This is going to be such a better experience IMO.

File: src/components/utils/hooks/useXmtpClient.tsx [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
 
+const ENCODING = 'binary';
+
+const buildLocalStorageKey = (walletAddress: string) => `xmtp:keys:${walletAddress}`;
+
+const loadKeys = (walletAddress: string): Uint8Array | null => {
+  const val = localStorage.getItem(buildLocalStorageKey(walletAddress));
+  return val ? Buffer.from(val, ENCODING) : null;
+};
+/**
```
Comment (bigint) [inline src/components/utils/hooks/useXmtpClient.tsx:16]:
```suggestion
};

/**
```

**Review by bigint — APPROVED** [review]
**LGTM** :shipit: 

Rad LGTM

**Pull request merged**